pkg/identity/identitysync/okta.go

```diff
@@ -133,28 +136,36 @@ func (o *OktaSync) ListUsers(ctx context.Context) ([]identity.IDPUser, error) {
 
 	log.Debugw("listing all okta users")
 
-	users, res, err := o.client.User.ListUsers(ctx, &query.Params{})
-	if err != nil {
-		// try and log the response body
-		logResponseErr(log, res, err)
-		return nil, errors.Wrap(err, "listing okta users from okta API")
-	}
-
-	log.Debugw("listed all okta users")
-
-	for res.HasNextPage() {
-		var nextUsers []*okta.User
-		res, err = res.Next(ctx, &nextUsers)
+	oktaUsers := make(map[string]*okta.User)
+	for _, groupId := range strings.Split(o.groups.Get(), ",") {
+		log.Infow(fmt.Sprintf("fetching users for groupId = %s", groupId))
+		users, res, err := o.client.Group.ListGroupUsers(ctx, groupId, &query.Params{})
 		if err != nil {
+			// try and log the response body
 			logResponseErr(log, res, err)
-			return nil, err
+			return nil, errors.Wrap(err, "listing okta users from okta API")
```

Comment:
Not sure here if we shouldn't bailout immediately and error on the first error.

Alternatively, we could acummulate the errors and return the aggregated list. 

but for the sake of KISS, it is better this way.

Reply:
Thanks for the note here @keymon. Agreed on failing early for KISS and also to retain consistency with our other IDP integrations which have this behaviour. In practice the most common failure mode we've observed is total failure of the IDP sync component due to misconfigured API keys